### PR TITLE
Add root_dir capability for multiple roots

### PR DESCRIPTION
**Pull request by djsegal**
---------------

Annotations for models in subdirectories of engines/subtrees/etc #291

---------------

I believe this is similar to the following issues: #190, #131, #88

To get it to work on my project, I have the following 2 lines in auto_annotate_models.rake:

```
 'model_dir'            => ["app/models", "lib/subtree/app/models"],
 'root_dir'             => ["/", "lib/subtree"],
```

### CODE REVIEW

File: lib/annotate.rb [hunk 1]
```diff
@@ -76,6 +76,10 @@ def self.setup_options(options = {})
       options[:model_dir] = ['app/models']
     end
 
+    if(options[:root_dir].empty?)
+      options[:root_dir] = ['/']
```
Comment (ctran) [inline lib/annotate.rb]: Is this supposed to be Rails.root? A slash doesn't seem to a correct default value from the caller/user perspective.

Comment (djsegal) [inline lib/annotate.rb]: + Thats true, we could just make the default `''`
+ The place where root_dir is used in will be unchanged:
```
File.join(root_directory, UNIT_TEST_DIR,  "%MODEL_NAME%_test.rb")
```

Comment (ctran) [inline lib/annotate.rb]: Could you make the change so I can accept the PR?

**Comment by djsegal** [comment]
@ctran I just finished the update. (i'll clean up the other PRs after the merge)

--------

One note, I did some minor refactoring:
+ I added `MATCHED_TYPES = %w(test fixture factory serializer scaffold)` to top of `lib/annotate/annotate_models.rb`
+ i default `get_patterns` to use `MATCHED_TYPES`, i.e. `get_patterns(pattern_types=MATCHED_TYPES)`

**Comment by ctran** [comment]
Awesome. Thanks.

**Pull request merged**